oximeter/db/src/client/mod.rs

```diff
@@ -72,18 +79,127 @@ mod probes {
     fn sql__query__done(_: &usdt::UniqueId) {}
 }
 
+// A "qorb connector" which creates a ReqwestClient for the backend.
+//
+// This also keeps track of the underlying address, so we can use it
+// for making HTTP requests directly to the backend.
+struct ReqwestConnector {}
+
+#[async_trait::async_trait]
+impl backend::Connector for ReqwestConnector {
+    type Connection = ReqwestClient;
+
+    async fn connect(
+        &self,
+        backend: &backend::Backend,
+    ) -> Result<Self::Connection, backend::Error> {
+        Ok(ReqwestClient {
+            client: reqwest::Client::builder()
+                .pool_max_idle_per_host(1)
+                .build()
+                .map_err(|e| QorbError::Other(anyhow!(e)))?,
+            url: format!("http://{}", backend.address),
+        })
+    }
+
+    async fn is_valid(
+        &self,
+        conn: &mut Self::Connection,
+    ) -> Result<(), backend::Error> {
+        handle_db_response(
+            conn.client
+                .get(format!("{}/ping", conn.url))
+                .send()
+                .await
+                .map_err(|err| QorbError::Other(anyhow!(err.to_string())))?,
+        )
+        .await
+        .map_err(|e| QorbError::Other(anyhow!(e)))?;
+        Ok(())
```

Comment:
Since we actually have a `ping` endpoint, I use it here to check connection liveness.